pilot/pkg/networking/v1alpha3/listener.go

```diff
@@ -406,10 +402,41 @@ type buildHTTPListenerOpts struct { // nolint: maligned
 	rds              string
 	useRemoteAddress bool
 	direction        http_conn.HttpConnectionManager_Tracing_OperationName
-	store            model.IstioConfigStore
 	authnPolicy      *authn.Policy
 }
 
+func buildDeprecatedHTTPListener(opts buildHTTPListenerOpts) *xdsapi.Listener {
+	if opts.rds != "" {
+		// Fetch V1 RDS response and stick it into the LDS response
+		rc, _ := v1.BuildRDSRoute(opts.env.Mesh, opts.proxy, opts.rds,
```

Comment:
Not sure I found the code where you generate the new style cluster names. 
It was actually working with RDSv1 (fetching), but was getting stuck not finding the expected clusters.

Reply:
Never mind, found it. Great hack !